homeassistant/components/matter/climate.py

```diff
@@ -0,0 +1,216 @@
+"""Matter climate platform."""
+from __future__ import annotations
+
+from typing import Any
+
+from chip.clusters import Objects as clusters
+from matter_server.client.models import device_types
+
+from homeassistant.components.climate import (
+    ClimateEntityDescription,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, Platform, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import MatterEntity
+from .helpers import get_matter
+from .models import MatterDiscoverySchema
+
+
+TEMPERATURE_SCALING_FACTOR = 100
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Matter climate platform from Config Entry."""
+    matter = get_matter(hass)
+    matter.register_platform_handler(Platform.CLIMATE, async_add_entities)
+
+
+class MatterClimate(MatterEntity, ClimateEntity):
+    """Representation of a Matter climate entity."""
+
+    features: int | None = None
+
+    _attr_temperature_unit: str = UnitOfTemperature.CELSIUS
+    _attr_supported_features: ClimateEntityFeature = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool mode."""
+        value = int(
+            self.get_matter_attribute_value(clusters.Thermostat.Attributes.SystemMode)
+        )
+        SystemMode = clusters.Thermostat.Enums.ThermostatSystemMode
+
+        match value:
+            case SystemMode.kAuto:
+                return HVACMode.AUTO
```

Comment:
The Home Assistant auto mode means the user can't control the temperature. Does it have the same meaning in Matter?

Reply:
Yes, it has the same meaning